src/native/profiler.cc

```diff
@@ -344,15 +345,24 @@ Profiler::runAgentThread(jvmtiEnv *jvmti_env, JNIEnv *jni_env, void *args) {
       jint end_line; //exclusive
       jint line = -1;
       std::vector<std::pair<jint, jint>> location_ranges;
+
+      bool select_last_line = true;
       for (int i = 1; i < num_entries; i++) {
         if (line == -1
             && entries[i].start_location > exp_frame.lineno) {
           line = entries[i - 1].line_number;
           current_experiment.lineno = line;
+          select_last_line = false;
           break;
         }
```

Comment:
What if we just did this?

```
if (i == num_entries || entries[i].start_location > exp_frame.lineno) {
      line = entries[i - 1].line_number;
      current_experiment.lineno = line;
}
```
I think that should handle the bug you pointed out and also simplify the existing logic.